### DISCUSSION TITLE
Prowler is taking too long to scan

### DISCUSSION THREAD

**Discussion by qadri99-max** [prowler-cloud/prowler#5304] [Q&A]
I am running Prowler in one region and scanning all services. The size of the account is moderate.

Prowler is taking almost 12 hours to finish the scan. I must be doing something wrong. 
I am running Prowler in ECS Fargate. I have increased ECS capacity to 4 vCPU and 12GB memory.

This is Prowler command:

prowler aws -z -f eu-west-1 --output-bucket my_bucket_name --output-directory my_dir --output-formats json-asff --security-hub role arn:aws:iam::$accountId:role/$ROLE_TO_ASSUME

Any suggestion what I can do? 

I am attaching a picture of logs which shows where its taking most time.

![security-hub-01](https://github.com/user-attachments/assets/ebebbd87-dc7e-4d1f-833a-bd5a07f87c8b)
Thanks

**Comment by jfagoagas** [comment]
Hello @qadri99-max I apologise for the late notice but this discussion wasn't notified from Github for some reason.

From the information you shared what is the service where you have the most resources? Because in Prowler there are several resources doing checks that could take a lot of time depending on the number of resources like checking secrets in AWS Lambda Code or AWS CloudWatch Log Groups.

There are several ways to speed up execution but those are based on your environment, like just scanning the regions you are using.

Thanks for using Prowler 🚀

**Reply by qadri99-max** [reply]
Thanks @jfagoagas 
After a bit more debugging, I can see the error messages are about Connection Timeout:
`Connect timeout on endpoint URL: "https://ec2.eu-west-1.amazonaws.com/"`

I am guessing the Subnet I am using for Fargate does not allow access to internet.

Is there any other VPC/Subnet settings I should be using?
Thanks

**Reply by jfagoagas** [reply]
Great catch! Prowler only needs access to the internet to get to the AWS APIs using always HTTPS, so to the 443 port.
